### DISCUSSION TITLE
Repo's philosophical future

### DISCUSSION THREAD

**Discussion by Paiusco** [offa/android-foss#194] [Ideas]
I have a couple of questions about the repo:
- How are the rules going to abide to the repo? Meaning, doing the `check_manually.sh` on the PR doesn't make more sense? Should a set of rules to be defined? (More than the `contributing.md`, for the user to know what kind of apps are here)
- Do you plan to add apps indefinitely? Maybe improve the categories sounds like a good idea then.
- I felt the need to enter each repo to get a better description of every app, it was very annoying, even more on generic categories, where it was not possible to get the idea of what the app is up to only by the name. What do you think? Any improvement here?

PS: I REALLY liked the repo's idea, congratz on that! Hope I can help making it better and bigger.

**Comment by offa** [comment]
> How are the rules going to abide to the repo? Meaning, doing the check_manually.sh on the PR doesn't make more sense? Should a set of rules to be defined? (More than the contributing.md, for the user to know what kind of apps are here)


Check manually was part of the CI checks previously, but it caused the builds to stall quite fast. For the world outside I may have looked like an attack and it got blocked :innocent: …

Do you have any rules in mind?

> Do you plan to add apps indefinitely? Maybe improve the categories sounds like a good idea then.

That's a good point … typically there are more apps added than removed.
Improving categories is a good way to keep overview. Once one section grows we can add subcategories for refinement too. #183 has some work in progress already.

None of them is carved in stone nor do all make perfect sense, so let's keep improving.

Also reviewing the listed entries from time to time and drop what's no longer beneficial.

> I felt the need to enter each repo to get a better description of every app, it was very annoying, even more on generic categories, where it was not possible to get the idea of what the app is up to only by the name. What do you think? Any improvement here?

I know, it's a balancing act between providing enough and too much information. Adding some description may help at some places but may add clutter at others. Piking a good text can be challenging too (project advertisement or "a flashlight / clock /…" vs. an spot on description).

What we can improve at a first place: Refine the current categories and try to get rid of the generic utilities and misc. 

> PS: I REALLY liked the repo's idea, congratz on that! Hope I can help making it better and bigger.

Thanks a lot! It's great to get feedback of any kind and quite some helping hands. You definitely do, so a huge thank you too!

**Comment by Paiusco** [comment]
> Check manually was part of the CI checks previously, but it caused the builds to stall quite fast. For the world outside I may have looked like an attack and it got blocked innocent …
Do you have any rules in mind?

Hum... I'll see if something comes up, however I couldn't find a better way to at least check if the project is archived (deprecated) using something cleaner.

>Also reviewing the listed entries from time to time and drop what's no longer beneficial.
Agreed. However, what you consider as 'no longer beneficial;? Not being actively developed is one, however having like 10 apps to read PDF doesn't seems like a good idea, otherwise in the end the repo will become F-droid's whole list :) 
In the other hand, if we measure it using github star we might end up killing new apps. Maybe a threshold? I'm thinking out loud here.


I also thought about a separate `.md` file for FOSS-beginners, containing a small list of apps maybe with a small description each, with the most famous one for them to start. (like signal for messaging, a browser, a couple of frontends...) like a tutorial on how to start, what do you think?
I don't want to trespass the repo's idea and start making it too big or too sparse over different ideas... so I rather ask :)

**Comment by offa** [comment]
> Hum... I'll see if something comes up, however I couldn't find a better way to at least check if the project is archived (deprecated) using something cleaner.

Checking for deprecation / archiving sounds interesting. However, maybe at this point it's a good idea to rewrite the script in something like python to prevent accessing every API from scratch.

> I also thought about a separate .md file for FOSS-beginners, containing a small list of apps maybe with a small description each, with the most famous one for them to start. (like signal for messaging, a browser, a couple of frontends...) like a tutorial on how to start, what do you think?

I'm a little sceptical of creating additional sublists / files. But what sounds interesting: A short tutorial on how to get started (eg. like the rooting one here).

While there's a list of apps, there's no real starting point on how to install, what tools are around (F-Droid, micro-G, custom ROMS), or where to get more information. There shouldn't be a duplication of information (eg. linking to the setup guides instead of rewriting them) but rather an answer to question like "I want to use FOSS on my device, where do I start?".
